client/landing/gutenboarding/stores/mapped-types.ts

```diff
@@ -0,0 +1,44 @@
+/**
+ * Mapped types
+ *
+ * This module should only contain mapped types, operations useful in the type system
+ * that do not produce any runtime code.
+ *
+ * Mapped types can be thought of as functions in the type system, they accept some type
+ * argument and transform it to another type.
+ *
+ * @see https://www.typescriptlang.org/docs/handbook/advanced-types.html#mapped-types
+ */
+
+/**
+ * Maps a "raw" selector object to the selectors available when registered on the wp-data store.
+ *
+ * @template S Selector map, usually from `import * as selectors from './my-store/selectors';`
+ */
+export type SelectFromMap< S extends Record< string, ( ...args: any[] ) => any > > = {
+	[ selector in keyof S ]: (
+		...args: TailParameters< S[ selector ] >
+	) => ReturnType< S[ selector ] >;
+};
+
+/**
+ * Maps a "raw" actions object to the selectors available when registered on the wp-data store.
+ *
+ * @template AS Selector map, usually from `import * as actions from './my-store/actions';`
+ */
+export type DispatchFromMap< AS extends Record< string, ( ...args: any[] ) => any > > = {
+	[ A in keyof AS ]: ( ...args: Parameters< AS[ A ] > ) => void;
```

Comment:
Total nitpick, but why `[ A in keyof AS ]` when for selectors we have `[ selector in keyof S] `? :grimacing: